react/components/EditorContainer/Topbar/ContextSelectors/index.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react'
+import { WithApolloClient, withApollo } from 'react-apollo'
+import { useIntl } from 'react-intl'
+import { ToastConsumerFunctions } from 'vtex.styleguide'
+
+import { useEditorContext } from '../../../EditorContext'
+import BindingSelector from './BindingSelector'
+import TenantInfoQuery from './graphql/TenantInfo.graphql'
+import LocaleSelector from './LocaleSelector'
+import { Binding, QueryStateReducer, TenantInfoData } from './typings'
+
+interface Props {
+  iframeRuntime: RenderContext
+  showToast: ToastConsumerFunctions['showToast']
+}
+
+const ContextSelectors: React.FC<WithApolloClient<Props>> = ({
+  client,
+  iframeRuntime,
+  showToast,
+}) => {
+  const [queryState, setQueryState] = React.useReducer<QueryStateReducer>(
```

Comment:
⛏ You don't actually need to `useReducer` here since it's possible to merge the previous state with a new state using `useState` by using a functional update (`setState(state => { ...state, newState })`), but both codes are easy to understand and maybe `useReducer` is "safer", so changing this is really optional.

Reply:
I believe it's easy to forget to explicitly preserve the previous state every time we use `setState`, while `useReducer` allows us to make it part of the "state updating contract", so that's why I chose the latter.